source/patterns/@aws-solutions-constructs/aws-lambda-sagemakerendpoint/lib/index.ts

```diff
@@ -0,0 +1,190 @@
+/**
+ *  Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import * as cdk from '@aws-cdk/core';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sagemaker from '@aws-cdk/aws-sagemaker';
+import * as defaults from '@aws-solutions-constructs/core';
+
+/**
+ * @summary The properties for the LambdaToSageMakerEndpoint class
+ */
+export interface LambdaToSageMakerEndpointProps {
+  /**
+   * Existing instance of Lambda Function object, if this is set then the lambdaFunctionProps is ignored
+   *
+   * @default - None
+   */
+  readonly existingLambdaObj?: lambda.Function;
+  /**
+   * User provided props to override the default props for the Lambda function
+   *
+   * @default - Default props are used
+   */
+  readonly lambdaFunctionProps?: lambda.FunctionProps;
+  /**
+   * Existing SageMaker Enpoint object, if this is set then the modelProps, endpointConfigProps, and endpointProps are ignored
+   *
+   * @default - None
+   */
+  readonly existingSageMakerEndpointObj?: sagemaker.CfnEndpoint;
+  /**
+   * User provided props to create SageMaker Model
+   *
+   * @default - None
+   */
+  readonly modelProps?: sagemaker.CfnModelProps;
+  /**
+   * User provided props to create SageMaker Endpoint Configuration
+   *
+   * @default - Default props are used
+   */
+  readonly endpointConfigProps?: sagemaker.CfnEndpointConfigProps;
+  /**
+   * User provided props to create SageMaker Endpoint
+   *
+   * @default - Default props are used
+   */
+  readonly endpointProps?: sagemaker.CfnEndpointProps;
+  /**
+   * An existing VPC for the construct to use (construct will NOT create a new VPC in this case)
+   *
+   * @default - None
+   */
+  readonly existingVpc?: ec2.IVpc;
+  /**
+   * Properties to override default properties if deployVpc is true
+   *
+   * @default - None
+   */
+  readonly vpcProps?: ec2.VpcProps;
+  /**
+   * Whether to deploy a new VPC
+   *
+   * @default - false
+   */
+  readonly deployVpc?: boolean;
+  /**
+   * Whether to deploy a natgatway in the new VPC (if deployVpc is true).
+   * If deployNatGateway is true, the construct creates Public and Private subnets.
+   * Otherwise, it creates Isolated subnets only
+   *
+   * @default - false
+   */
+  readonly deployNatGateway?: boolean;
```

Comment:
NAT Gateways are controlled through the VPC Props - breaking just this VPC prop functionality out into a separate attribute on the construct props doesn't seem to make sense. Do we have a compelling reason?

(eg - other constructs may also want a NAT Gateway, but for those the client would use VPC Props. But all constructs should work the same way unless there is a reason why they can't).

Reply:
The reason I had to create the `deploNatGateway?` property is that the `natGateways` and `subnetConfiguration` are controlled by the construct, not `vpcProps?`, according to the description of `vpcProps?` in the `aws-lambda-sqs` construct. 
```
| |vpcProps?|ec2.VpcProps|Optional user-provided properties to override the default properties for the new VPC. enableDnsHostnames, enableDnsSupport, natGateways and subnetConfiguration are set by the pattern, so any values for those properties supplied here will be overrriden. If deployVpc is not true then this property will be ignored.
```